hir/src/globals.rs

```diff
@@ -164,26 +189,26 @@ impl GlobalVariableTable {
         //
         // At the end, the effective address of the pointer is the total
         // size in bytes of the allocation
-        let mut hp = 0 as *const u8;
+        let mut size = 0;
         for gv in self.layout.iter() {
             let layout = gv.layout();
-            // SAFETY: We aren't actually using these pointers,
-            // we're just making use of the optimized alignment
-            // intrinsics available for them
-            let offset = hp.align_offset(layout.align());
-            unsafe {
-                hp = hp.add(offset).add(layout.size());
+            let align = layout.align();
+            let mut gv_size = layout.size();
+            let align_offset = size % align;
+            if align_offset != 0 {
+                gv_size += align - align_offset;
             }
```

Comment:
The same calculation is replicated in `offset_of`. It could probably be a helper function.